src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,22 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private static final Map<String, UserData> cache = new ConcurrentHashMap<>();
     private final Executor executor;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (!cache.containsKey(userId)) {
+            cache.put(userId, new UserData(userId, "Details for " + userId));
```

Comment:
Everything fine but your task also provides  delay imitation ;) `The processing can be a simulated task, like fetching user data from a database (mock this with a delay).`